pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/TransactionTest.java

```diff
@@ -828,4 +831,30 @@ public void testNotChangeMaxReadPositionAndAddAbortTimesWhenCheckIfNoSnapshot()
         Assert.assertEquals(changeMaxReadPositionAndAddAbortTimes.get(), 0L);
 
     }
+
+    @Test
+    public void testAutoCreateSchemaForTransactionSnapshot() throws Exception {
+        String namespace = TENANT + "/ns2";
+        String topic = namespace + "/test";
+        pulsarServiceList.get(0).getConfiguration().setAllowAutoUpdateSchemaEnabled(false);
```

Comment:
The request might point to other brokers, only disabling schema auto-update for 1 broker will not cover the changes well, you can try to use the admin API to disable the schema auto-update or disable all the brokers.